src/analysis/plugin/plugin.py

```diff
@@ -145,7 +149,7 @@ def get_analysis(self, file_handle: io.FileIO, virtual_file_path: dict, analyses
             'system_version': self.metadata.system_version,
             'summary': summary,
             'tags': tags_dict,
-            'result': result.dict() if result else None,
+            'result': result.dict() if isinstance(result, BaseModel) else None,
```

Comment:
Why is this needed?

Reply:
I think during the time I wasn't sure if the plugin always returns a `BaseModel` or if it can be a `dict` if no schema is specified. I think as it is now, the schema is mandatory and the plugin always returns a `BaseModel`, right?

Reply:
> right?

Yes!